neuralpp/experiments/experimental_inference/exact_belief_propagation.py

```diff
@@ -35,18 +37,24 @@ def __init__(self, factors, query):
         super().__init__(LazyFactorSpanningTree(FactorGraph(factors), query))
 
 
+Expansion = namedtuple("Expansion", "node expansion_value")
+
+
 class AnytimeExactBeliefPropagation(TreeComputation):
```

Comment:
Like I mentioned in our meeting, `AnytimeExactBeliefPropagation` is taking care of both probabilistic inference and managing the mechanics of expanding a `TreeComputation` and updating it. Would it be possible to define a `PartialTreeComputation` class that subclasses `TreeComputation` and takes care of expanding/updating values? Then `AnytimeExactBeliefPropagation` would just define the message computations.